### PR TITLE
Log cache hit/miss for deepseek

### PR DESCRIPTION
**Pull request by enyst**
**End-user friendly description of the problem this fixes or functionality that this introduces**

- [ ] Include this change in the Release Notes. If checked, you must provide an **end-user friendly** description for your change below
Display in the logs the prompt cache hit or miss for Deepseek Chat.

---
**Give a summary of what the PR does, explaining any non-trivial design decisions**

Just a small tweak, display in the logs the cache hit/miss for Deepseek as we are doing for Anthropic.

---
**Link of any specific issues this addresses**

### CODE REVIEW

**Comment by enyst** [comment]
@xingyaoww you're right, I changed to use this object for cache hits. Then I left only in Anthropic the cache writes, because unlike other providers, these are 25% more expensive iirc. So people may want to see that information.

(the Anthropic cache write tokens are not the exact difference between the other values, either)

Deepseek:
image

Anthropic models:
image

File: openhands/llm/llm.py [hunk 1]
```diff
@@ -291,19 +291,21 @@ def _post_completion(self, response: ModelResponse) -> None:
...
-                stats += (
-                    'Input tokens (cache read): ' + str(cache_read_input_tokens) + '\n'
-                )
+            # NOTE: the keys are different for Anthropic and DeepSeek
+            # Anthropic: cache_creation_input_tokens, cache_read_input_tokens
+            # DeepSeek: prompt_cache_miss_tokens, prompt_cache_hit_tokens
+
+            cache_write_tokens = model_extra.get(
+                'cache_creation_input_tokens'
+            ) or model_extra.get('prompt_cache_miss_tokens')
```
[the remark this answers is not in the document]
Comment (enyst) [reply openhands/llm/llm.py]: Done!

**Comment by xingyaoww** [comment]
@enyst Is it possible to calculate cache write by doing `total_tokens - cached_tokens` so we don't need to handle this specifically for anthropic?

**Comment by enyst** [comment]
> @enyst Is it possible to calculate cache write by doing `total_tokens - cached_tokens` so we don't need to handle this specifically for anthropic?

It's not _exactly_ that difference. 😅 Anthropic doesn't write to cache automatically, we need to tell it what to cache and what to leave alone. And we do: we always leave out _at least_ this little bit that would result in cache misses all the time if it was included: [funny variable bit of context](https://github.com/All-Hands-AI/OpenHands/blob/a3c49538fca459c93cbb8c655bf4d360601c2029/openhands/agenthub/codeact_agent/codeact_agent.py#L277).

I honestly don't think it's worth bothering with the reminder at this point, since well, we are likely heading towards a lot more variance in the context so we'll have more cache-missing stuff to handle.

**Comment by enyst** [comment]
In other words, unlike Deepseek, Anthropic has 3 costs for input tokens:
- cache writes (iirc +25% regular cost)
- cache reads  (10% of regular cost)
- regular input tokens (regular cost)

We need to use all 3 currently, because of our reminder.

**Pull request merged**